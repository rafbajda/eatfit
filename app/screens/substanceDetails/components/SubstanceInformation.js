import React from 'react';
import { Text } from 'native-base';
import {
    SubstanceInformationContainer,
    SubstanceInformationLabel,
} from '../styles/substanceDetailsStyles';
import {
    ScanInformationContainer,
    ScanInformationDescription,
    ScanInformationInfo,
    ScanRate,
    ScanRatingIcon
} from "../../scanDetails/styles/scanDetailsStyles";
import hps from "../../scanDetails/utils/helpers";

const SubstanceInformation = props => {
    const { substanceName, score } = props;
    const roundedScore = Math.round(score * 100) / 100;
    const emoji = hps.getEmoji(score);
    const color = hps.getColor(score);
    const description = hps.getDescription(score);
    return (
        <SubstanceInformationContainer>
            <SubstanceInformationLabel>Name:</SubstanceInformationLabel>
            <Text>{substanceName}</Text>
            <ScanInformationInfo>
                <ScanRate color={color}>{roundedScore}</ScanRate>
                <ScanRatingIcon>{emoji}</ScanRatingIcon>
            </ScanInformationInfo>
            <ScanInformationDescription color={color}>
                {description}
            </ScanInformationDescription>
        </SubstanceInformationContainer>
    );
};

export default SubstanceInformation;
