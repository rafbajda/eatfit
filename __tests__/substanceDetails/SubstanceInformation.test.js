import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import SubstanceInformation from '../../app/screens/substanceDetails/components/SubstanceInformation';

describe('SubstanceInformation snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<SubstanceInformation />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
