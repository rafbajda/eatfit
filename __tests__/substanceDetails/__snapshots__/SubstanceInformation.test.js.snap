// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubstanceInformation snapshot renders component 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  className="sc-htpNat KtFsv"
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": "10%",
      },
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <Text
      className="sc-bxivhb czhwhJ"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      [missing "en.labels.name" translation]
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    />
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      className="sc-gzVnrw ibZhkv"
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <Text
          className="sc-dnqmqq jalEwm"
          color="#00FF00"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          NaN
        </Text>
        <Text
          className="sc-htoDjs jtcSTs"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          üòÅ
        </Text>
      </View>
    </RCTScrollView>
    <Text
      className="sc-iwsKbI jOpSwg"
      color="#00FF00"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
    >
      [missing "en.healthImpactLabels.veryGood" translation]
    </Text>
  </View>
</RCTScrollView>
`;
